### PR TITLE
Fix OS X get_ticks_usec return value after #22424

### LINKED ISSUE #22564: Time scale is incorrect in Mac
**Godot version:**

Master (3ee657e)
MacOSX

**Issue description:**

Time scale is incorrect in Mac

Fps: always display 1
Process: display 16.XX

(it is 60 FPS before)

and input cursor flash very quick

3D view movement is very quick too (Mouse Right Click + WSAD)

**Steps to reproduce:**
use master version
do movement in 3D view
or try to input in line edit

### PR DESCRIPTION
**Pull request by marcelofg55**
Fixes #22564.

### CODE REVIEW

File: drivers/unix/os_unix.cpp [hunk 1]
```diff
@@ -74,7 +74,7 @@ static void _setup_clock() {
 	kern_return_t ret = mach_timebase_info(&info);
 	ERR_EXPLAIN("OS CLOCK IS NOT WORKING!");
 	ERR_FAIL_COND(ret != 0);
-	_clock_scale = (double)info.numer / (double)info.denom;
+	_clock_scale = (double)info.numer / (double)info.denom / 1000.0;
```
Comment (akien-mga) [inline drivers/unix/os_unix.cpp]: Parentheses would be nice here to make the intent clear.

Comment (marcelofg55) [reply drivers/unix/os_unix.cpp]: Added

Comment (Faless) [reply drivers/unix/os_unix.cpp]: I think `(double)info.numer / (double)(info.denom * 1000)` might be better for precision.
as `info.denom` is an integer.

**Review by akien-mga — APPROVED** [review]

**Comment by Faless** [comment]
I think `(double)info.numer / (double)(info.denom * 1000)` might be better for precision.
as `info.denom` is an integer.

**Comment by Faless** [comment]
> I think `(double)info.numer / (double)(info.denom * 1000)` might be better for precision.
> as `info.denom` is an integer.

On a second thought, that has the downside that it might overflow.
I don't want to overcomplicate this, especially since my knowledge of numerical errors is not so great.
And we might do even worse when we do the other cast to double.
I think this can be merged as is after all.

**Comment by Faless** [comment]
Thanks :+1:

**Pull request merged**